src/bin/read_csv/mod_dta.c

```diff
@@ -387,39 +386,98 @@ static readstat_value_t value_double_dta(const char *s, size_t len, struct csv_m
 }
 
 static readstat_value_t value_double_date_time_dta(const char *s, size_t len, struct csv_metadata *c) {
-    char *dest;
-    readstat_variable_t *var = &c->variables[c->columns];
-    double val = strtod(s, &dest);
-    if (dest == s) {
-        fprintf(stderr, "not a number: %s\n", (char*)s);
+    // Handle empty or NULL strings as missing values
+    if (s == NULL || len == 0 || *s == '\0') {
+        readstat_value_t value = {
+            .type = READSTAT_TYPE_DOUBLE,
+            .is_system_missing = 1,
+            .v = { .double_value = NAN }
+        };
+        return value;
+    }
+
+    // Truncate the date string to 23 characters to remove the timezone offset and
+    // microseconds, if present. STATA does not support timezones or microseconds.
+    char date_time[24];
+    strncpy(date_time, s, 23);
+    date_time[23] = '\0';
+
+    // Parse date-time components
+    int year, month, day, hour, minute, second, msecs = 0;
+    int matched = sscanf(
+        date_time,
+        "%d-%d-%d %d:%d:%d.%d",
+        &year, &month, &day, &hour, &minute, &second, &msecs
+    );
+    if (matched < 6 || matched > 8) {
+        fprintf(stderr, "%s:%d not a valid date-time: %s (expected format: yyyy-mm-dd hh:MM:SS with optional milliseconds. Datetime string is truncated at 23 characters to ignore microseconds and timezone information.)\n", __FILE__, __LINE__, date_time);
         exit(EXIT_FAILURE);
     }
-    int missing_ranges_count = readstat_variable_get_missing_ranges_count(var);
```

Comment:
Why was this logic removed?

Reply:
I had initially copied the `value_double_dta` to `value_double_date_time_dta` to ensure code was being reached, before implementing `value_double_date_time_dta` as its own function. I wasn't sure how this code would handle blank date fields in the CSV.

Reply:
OK, I'm guessing missing ranges are seldom used with dates, so let's leave it out